lib/Dialect/Torch/IR/TorchOps.cpp

```diff
@@ -763,6 +781,35 @@ void AtenLenTOp::getCanonicalizationPatterns(RewritePatternSet &patterns,
   });
 }
 
+//===----------------------------------------------------------------------===//
+// AtenAddTensorOp
+//===----------------------------------------------------------------------===//
+
+void AtenAddTensorOp::getCanonicalizationPatterns(RewritePatternSet &patterns,
+                                                  MLIRContext *context) {
+  patterns.add(+[](AtenAddTensorOp op, PatternRewriter &rewriter) {
+    // The lhs and rhs of the add.tensor op should be 0d tensors for the
+    // canonicalization to be carried out.
+    // `aten.add.tensor(self, other, alpha)` is canonicalized to
+    // `aten.add.int(self, aten.mul.int(other, alpha))`.
+
+    llvm::Optional<Value> lhs = getScalarValue(op, rewriter);
+
+    if (!lhs.hasValue())
+      return failure();
+
+    auto rhsScalar = op.other().getDefiningOp<PrimNumToTensorScalarOp>();
+    if (!rhsScalar)
+      return failure();
+    Value rhs = rhsScalar.a();
+    Value mul = rewriter.create<AtenMulIntOp>(op->getLoc(), rhs, op.alpha());
```

Comment:
do you have to check that the datatypes are actually ints?

Reply:
@silvasean, I added type checking for `mul` and `add` variables

Reply:
Update instead of `mul` and `add` type checking, I did type checking for `lhs` and `rhs`